### PR TITLE
Wait... are those razor blades? (Adds razor whip and razor blade recipe.)

### PR DESCRIPTION
**Pull request by Greevv**
#### Summary
```SUMMARY: Content "Adds a new weapon, the razor whip, made from razor blades and leather."```

#### Purpose of change
I wanted to add a whip that was actually strong enough to be usable, but that could be created from relatively easily obtained materials.

#### Describe the solution
I added a new weapon, the razor whip. The razor whip requires 20 leather and 5 razor blades to make, in addition to 8 Fabrication. This would be another mid-lategame weapon option.

I also added a recipe for razor blades that is unlocked fairly early. Additionally, razor blades now have some tool qualities so you can use them as a decent butchering tool.

I also added the razor whip, chain, and bullwhip to the list of Silat compatible weapons, as Silat historically employed whips and chains. ( ͡° ͜ʖ ͡°)

#### Additional context
Sources:
[Weapons of Silat](https://en.wikipedia.org/wiki/Weapons_of_silat#Flail_weapons)

### CODE REVIEW

File: data/json/recipes/other/tool.json [hunk 1]
```diff
@@ -177,5 +177,26 @@
...
     "qualities": [ { "id": "CUT", "level": 1 } ],
     "components": [ [ [ "can_food_unsealed", 2 ], [ "scrap", 4 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "result": "razor_blade",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "skills_required": [ "fabrication", 3 ],
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: No need to specify same skill twice in skill_used in skills_required.

File: data/json/recipes/other/tool.json [hunk 2]
```diff
@@ -177,5 +177,26 @@
...
+    "autolearn": true,
+    "book_learn": [
+      [ "mag_survival", 1 ],
+      [ "textbook_survival", 1 ],
+      [ "mag_fabrication", 1 ],
+      [ "manual_fabrication", 1 ],
+      [ "textbook_fabrication", 1 ]
+    ],
+    "qualities": [ { "id": "HAMMER", "level": 1 }, { "id": "SAW_M", "level": 1 } ],
+    "tools": [ [ [ "fire", -1 ] ] ],
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: Would need forging tools; steel can't be melted by fire.

File: data/json/recipes/other/tool.json [hunk 3]
```diff
@@ -177,5 +177,26 @@
...
+    "components": [ [ [ "scrap", 1 ], [ "can_food_unsealed", 1 ] ] ]
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: Tin can wouldn't work for that; blade needs to be made from steel.

File: data/json/items/melee.json [hunk 4]
```diff
@@ -1443,7 +1443,24 @@
     "price_postapoc": 100,
     "material": "steel",
     "weight": 10,
-    "cutting": 1
+    "cutting": 1,
+    "qualities": [ [ "BUTCHER", 9 ], [ "CUT_FINE", 1 ], [ "CUT", 1 ] ]
```
Comment (Rail-Runner) [inline data/json/items/melee.json]: Seems too small and/or awkward to qualify for any of those qualities.

File: data/json/recipes/other/tool.json [hunk 5]
```diff
@@ -177,5 +177,26 @@
...
+  },
+  {
+    "type": "recipe",
+    "result": "razor_blade",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "skills_required": [ "fabrication", 3 ],
+    "difficulty": 3,
+    "time": 8000,
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: If it's forged, would need much more time to craft it.

File: data/json/recipes/other/tool.json (hunk 2 above)
Comment (Greevv) [reply data/json/recipes/other/tool.json]: No forging is being done. The steel is being mostly cut to shape.

File: data/json/recipes/other/tool.json (hunk 3 above)
Comment (Greevv) [reply data/json/recipes/other/tool.json]: Modern food cans are made from steel, not tin.

File: data/json/items/melee.json (hunk 4 above)
Comment (Greevv) [reply data/json/items/melee.json]: I respectfully disagree, I've used razor blades as a tool plenty of times.

File: data/json/recipes/other/tool.json (hunk 5 above)
Comment (Greevv) [reply data/json/recipes/other/tool.json]: Again, not forged.

File: data/json/recipes/other/tool.json (hunk 1 above)
Comment (Greevv) [reply data/json/recipes/other/tool.json]: `skill_used` decides what skill is trained and used for success checks, `skills_required` defines what skill level the recipe is unlocked at.

**Comment by RarkGrames** [comment]
If the damage of a razorblade is 1, and the damage of the whip is 3 by default, where does the 22 damage come from? Also, to outfit a 2-3 meter long bullwhip with razorblades, wouldn't you need around 50-75 razor blades (each being around 40 mm long)?

**Comment by Greevv** [comment]
I mean, taking the sum of components isn't exactly a good way to get a good damage value. Also, as for the number of razor blades, yeah, it probably should have more. As is, I think the bullwhip is way underpowered. Would have changed the damage value on it but I was trying to keep the number of changes in one PR minimal.

**Comment by RarkGrames** [comment]
The closest real life equivalent to this is the barbed whip used by slave masters on galleys and other places in ancient times. The main purpose of it was to cause shallow open wounds to inflict as much pain as possible without actually killing the slave or damaging his body.

The 22 damage is halfway between combat knife (18) and cavalry sabre (26). That is already into the "cut someone's stomach open and gut them like a fish" territory.

**Comment by Greevv** [comment]
As you said, the main purpose was to cause superficial wounds. That whip was designed with that in mind. This whip was designed for killin'. Also, I can't seem to find a source that barbed whips were ever a thing, care to provide a source? The closest real life equivalent I've found is the Russian "knout", which often did kill its victims.

**Comment by RarkGrames** [comment]
On a quick search on " barbed cat-o-nine tails" I had a lot of whatever hits, but here's one reliable one:

https://www.britannica.com/topic/Sing-Sing#ref1234284

**Comment by Greevv** [comment]
While a cat o' nine tails is certainly a type of whip, it's nothing at all
like what I was going for with this.

**Comment by I-am-Erk** [comment]
I am deeply, deeply skeptical that this could reasonably be used without slicing your face off.  It seems like a *more* uncontrollable [urumi](https://en.wikipedia.org/wiki/Urumi).

Perhaps the combination of razors and ropes would be better employed to make clothesline traps to behead zombies?

**Comment by Night-Pryanik** [comment]
This definitely would a be a weapon that require A LOT of training to operate and not to damage yourself. I can't imagine a person with 0-1 levels in melee effectively fighting with this weapon.

**Comment by eilaattwood** [comment]
> I am deeply, deeply skeptical that this could reasonably be used without slicing your face off. It seems like a _more_ uncontrollable [urumi](https://en.wikipedia.org/wiki/Urumi), but without all the benefits of spring steel.
I think, spring steel is actually the _source_ of lack of control in urumi. Because spring elements can redirect applied forces, while the flexible whip can't.

**Comment by Greevv** [comment]
Is there a way to make it require a minimum melee level to wield?

**Comment by Night-Pryanik** [comment]
> Is there a way to make it require a minimum melee level to wield?

I believe there isn't.

**Comment by I-am-Erk** [comment]
Spring steel may make it even harder (the urumi is considered one of the hardest weapons to use in the world from what I've read), but I can't find a single combination of whips and edges that is referred to as anything but difficult to learn and easy to hurt yourself with. Strapping razors to a whip seems like a recipe for deep cuts more than damage to a zombie.

Like I said though, razor blades on a rope would make a wonderful trap. Zombies are too stupid to do anything but push through it.

**Comment by Greevv** [comment]
I agree that the razor wire trap is a good idea, and I'd even be willing to try and implement it, but it doesn't accomplish what I set out to do at all.

**Comment by eilaattwood** [comment]
> Spring steel may make it even harder (the urumi is considered one of the hardest weapons to use in the world from what I've read), but I can't find a single combination of whips and edges that is referred to as anything but difficult to learn and easy to hurt yourself with. Strapping razors to a whip seems like a recipe for deep cuts more than damage to a zombie.
> 
> Like I said though, razor blades on a rope would make a wonderful trap. Zombies are too stupid to do anything but push through it.

Yep, such weapon will be high pain, high bleeding chance, but low damage. So, almost useless against zombies...

**Comment by Greevv** [comment]
Closing this, too much dissent. Will rework things and submit a new PR.

**Pull request closed without merging**